projects/storefrontlib/shared/components/assistive-technology-message/assistive-technology-message.directive.ts

```diff
@@ -0,0 +1,48 @@
+import {
+  Directive,
+  ElementRef,
+  HostListener,
+  Input,
+  Optional,
+  TemplateRef,
+} from '@angular/core';
+import { GlobalMessageService } from '../../../../core/src/global-message/facade/global-message.service';
+import { GlobalMessageType } from '../../../../core/src/global-message/models/global-message.model';
+
+@Directive({
+  selector: '[cxAtMessage]',
+})
+export class AtMessageDirective {
+  /**
+   * Usage [cxAtMEssage]="'translatableKey' | cxTranslate"
+   */
+  @Input() cxAtMessage: string;
+
+  constructor(
+    protected elementRef: ElementRef<HTMLElement>,
+    @Optional() protected templateRef: TemplateRef<HTMLElement>,
+    protected globalMessageService: GlobalMessageService
+  ) {}
+
+  protected get host(): HTMLElement {
+    return !!this.templateRef
+      ? this.templateRef.elementRef.nativeElement.parentElement
+      : this.elementRef.nativeElement;
+  }
+
+  /**
+   * Emit assistive global meesage to improve screen reader vocalization.
+   * @param event
+   */
+  @HostListener('click', ['$event'])
+  handleClick(event: MouseEvent): void {
+    event?.preventDefault();
+
+    if (event?.target === this.host) {
+      this.globalMessageService.add(
+        this.cxAtMessage,
```

Comment:
if `this.cxAtMessage` is empty (`undefined`), we can get the message just from `innerText`.

Reply:
@WeizhengSap I am not too sure if this is needed for the reason that the entire purpose of this extension of global message is to make screen reader technologies vocalize a descriptive text that is not on the DOM. So the usage strictly requires passing a descriptive message. 

I do see how it would serve as a fallback but the innerText of the element is not the intended message. Thoughts?

Reply:
OK, if `cxAtMessage` is empty, do nothing. (I thought we need do check of `cxAtMessage` to prevent empty key/text passed to globalMessage.add)